dlt/destinations/weaviate/weaviate.py

```diff
@@ -0,0 +1,370 @@
+            )
+            self._execute_schema_update(only_tables)
+        else:
+            logger.info(
+                f"Schema with hash {self.schema.stored_version_hash} "
+                f"inserted at {schema_info.inserted_at} found "
+                f"in storage, no upgrade required"
+            )
+
+        return applied_update
+
+    def _execute_schema_update(self, only_tables: Iterable[str]) -> None:
+        for table_name in only_tables or self.schema.tables:
+            table = self.schema.tables[table_name]
+            class_schema = self.make_weaviate_class_schema(table)
+
+            self.db_client.schema.create_class(class_schema)
+        self._update_schema_in_storage(self.schema)
+
+    def get_schema_by_hash(self, schema_hash: str) -> Optional[StorageSchemaInfo]:
+        version_class_name = table_name_to_class_name(VERSION_TABLE_NAME)
+
+        try:
+            self.db_client.schema.get(version_class_name)
+        except weaviate.exceptions.UnexpectedStatusCodeException as e:
+            if e.status_code == 404:
+                return None
+            raise
+
+        properties = [
+            "version_hash",
+            "schema_name",
+            "version",
+            "engine_version",
+            "inserted_at",
+            "schema",
+        ]
+
+        response = (
+            self.db_client.query.get(version_class_name, properties)
+            .with_where(
+                {
+                    "path": ["version_hash"],
+                    "operator": "Equal",
+                    "valueString": schema_hash,
+                }
+            )
+            .with_limit(1)
+            .do()
+        )
+
+        try:
+            record = response["data"]["Get"][version_class_name][0]
+        except IndexError:
+            return None
+        return self._decode_schema(record)
+
+    def _decode_schema(self, record: Dict[str, Any]) -> StorageSchemaInfo:
+        # XXX: Duplicate code from dlt/destinations/job_client_impl.py
+        schema_str = record["schema"]
+        try:
+            schema_bytes = base64.b64decode(schema_str, validate=True)
+            schema_str = zlib.decompress(schema_bytes).decode("utf-8")
+        except binascii.Error:
+            pass
+
+        return StorageSchemaInfo(
+            version_hash=record["version_hash"],
+            schema_name=record["schema_name"],
+            version=record["version"],
+            engine_version=record["engine_version"],
+            inserted_at=pendulum.parse(record["inserted_at"]),
+            schema=schema_str,
+        )
+
+    def make_weaviate_class_schema(self, table: TTableSchema) -> Dict[str, Any]:
+        """Creates a Weaviate class schema from a table schema."""
+        table_name = table["name"]
+
+        class_name = table_name_to_class_name(table_name)
+
+        if is_dlt_table(table_name):
+            return self._make_non_vectorized_class_schema(class_name, table)
```

Comment:
Good point, the difference is that for `_make_non_vectorized_class_schema()` I skip vectorization on the class level versus property level in `_make_vectorized_class_schema()` case:

```
return {
    "class": class_name,
    "properties": properties,
    "vectorizer": "none",
    "vectorIndexConfig": {
        "skip": True,
    },
}
```

@hsm207 is there any difference in terms of internal implementation?

Reply:
@burnash if in the a class, the user sets:

```
"vectorIndexConfig": {
        "skip": True,
    }
```

then, it is recommended to also set `"vectorizer"` to `"none"` in the class too. Otherwise, the vectorizer will still run to create the embeddings but will not index it.

Reply:
@hsm207 good point, that's what I'm currently doing for the classes that skip vectorization entirely (see the `return` above). My question is what's the difference between skipping vectorization for every property versus skipping it on the class level (both with `"vectorizer": "none"` and `"vectorIndexConfig": {"skip": True}`)

Reply:
@burnash skipping vectorization for every property is not equivalent to `"vectorizer": "none"` and `"vectorIndexConfig": {"skip": True}`. In the former, the text2vec module would still be executed to check whether to create an embedding for the object. In the latter, this check is skipped entirely.